packages/tools/src/utilities/pointInSurroundingSphereCallback.ts

```diff
@@ -4,33 +4,134 @@ import type { Types } from '@cornerstonejs/core';
 import type { vtkImageData } from '@kitware/vtk.js/Common/DataModel/ImageData';
 import { vec3 } from 'gl-matrix';
 import { pointInSphere } from './math/sphere';
-import { getBoundingBoxAroundShape } from './segmentation/getBoundingBoxUtils';
 import pointInShapeCallback, {
   PointInShapeCallback,
 } from './pointInShapeCallback';
+import { BoundsIJK } from '../types';
+import { getBoundingBoxAroundShape } from './boundingBox';
 
 const { transformWorldToIndex } = csUtils;
 
-// Todo: I *think* this can be done without the need to access viewport's camera
-// since sphere's center circle can be in any plane as long as its center
-// is the center of the sphere ..
-
 /**
- * Given a viewport, an imageData, and a circle points in the viewport, it will
- * run the callback for each point in sphere whose great circle (biggest circle
- * that can be drawn chopping a sphere) is the provided circle points.
+ * Given an imageData, and the great circle top and bottom points of a sphere (
+ * great circle also known as orthodrome is the intersection of the sphere and
+ * the plane that passes through the center of the sphere), this function will
+ * run the callback for each point of the imageData that is within the sphere
+ * defined by the great circle points.
  *
- * @param viewport - VolumeViewport
  * @param imageData - The volume imageData
- * @param circlePoints - [Types.Point3, Types.Point3]
+ * @param circlePoints - bottom and top points of the great circle in world coordinates
  * @param callback - A callback function that will be called for each point in the shape.
  */
 export default function pointInSurroundingSphereCallback(
-  viewport: Types.IVolumeViewport,
   imageData: vtkImageData,
   circlePoints: [Types.Point3, Types.Point3],
-  callback: PointInShapeCallback
+  callback: PointInShapeCallback,
+  viewport?: Types.IVolumeViewport
 ): void {
+  // We can run the sphere equation to determine if a point is inside
```

Comment:
Is an estimation good enough? I don't get how working this out in ijk space is useful when these dimensions could be e.g. (1,1,3) in many very common cases, you need to do all these calculations in world space, no?